tools/viewer/main.rs

```diff
@@ -164,6 +171,7 @@ fn main() -> Result<()> {
 
 fn init_compiler(
     args: &Cli,
+    watched_paths: Arc<Mutex<HashMap<PathBuf, ()>>>,
```

Comment:
Rust also have a HashSet

Also we can clear it at the beginning of this function so that path that were watched from a previous run are no longer in it.  (but we can add the main file in it there)